src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -834,7 +897,7 @@ public override int Read(Span<byte> destination)
                         throw new IOException(SR.net_http_client_execution_error, http2Stream._abortException);
                     }
 
-                    return http2Stream.ReadData(destination, CancellationToken.None);
+                    return http2Stream.ReadData(destination, _responseMessage);
```

Comment:
Not sure it matters, but should we clear _responseMessage when we hit EOF and process the trailers?

We'd probably need to change the logic here to check for EOF here instead of in the callee. Not sure it's worth it.

Reply:
I'd considered that, but it didn't seem worthwhile.  If you can think of a strong reason that doing so is a good idea, let me know, but otherwise I'll leave it as-is.

Reply:
Ok by me